### ISSUE TITLE
No spamScore or baseSpamScore

### ISSUE THREAD

**Issue by CliffS** [schahriar/spamc-stream#3]
I am using this library in an SMTP program.

The email message is in a buffer `data` and I am calling the library as follows:

          Spamc = require('spamc-stream');

          spamc = new Spamc;
          reporter = spamc.symbols({
            'Content-length': data.length
          });
          stream = new Stream.PassThrough;
          stream.end(data);
          stream.pipe(reporter);
          return reporter.on('report', function(report) {
            console.log('REPORTER', report);

Mostly I get back structures like this:

```
REPORTER { responseCode: 0,
  responseMessage: 'EX_OK',
  isSpam: false,
  spamScore: 4.7,
  baseSpamScore: 5,
  matches: 
   [ 'DKIM_SIGNED',
     'DKIM_VALID',
     'DKIM_VALID_AU',
     'RCVD_IN_PBL',
     'RCVD_IN_SORBS_DUL' ] }
```

but quite often I see returns like:

```
REPORTER { responseCode: 0,
  responseMessage: 'EX_OK',
  matches: 
   [ 'DKIM_SIGNED',
     'DKIM_VALID',
     'DKIM_VALID_AU',
     'HTML_FONT_LOW_CONTRAST',
     'HTML_MESSAGE',
     'NO_RECEIVED' ] }
```

Am I doing something wrong?

**Comment by schahriar** [comment]
If you are passing the data as a buffer instead of a stream I suggest changing the order of the function a little bit to this:

```javascript
...
stream = new Stream.PassThrough;
stream.pipe(reporter);
stream.end(data);
...
```

However, I highly doubt that is your issue, the root cause for this would probably be within spamc server itself or with the way it was setup/trained on your side. I suggest that you capture an example message that leads to no `spamScore` header and manually run it under the spamc CLI [https://spamassassin.apache.org/full/3.1.x/doc/spamc.html](https://spamassassin.apache.org/full/3.1.x/doc/spamc.html) to see if you can reproduce the results.

**Comment by CliffS** [comment]
`stream.end()` is always async and will wait until there is a reader, in this case `stream.pipe()`, so that will not be a problem.

Actually, since writing the original report, I've realised that your regex doesn't allow for negative spam scores.

Also the regex for collecting the matches will always miss the last one as there is no final comma in the input.

**Comment by schahriar** [comment]
@CliffS you are right, did not know that. The regular expressions came from the original author (https://github.com/coxeh/node-spamc) with a few modifications on top, I don't have enough test cases to dig in but if you are willing to make a PR let me know.
